src/WebJobs.Script/Config/FunctionsHostingConfigProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Permissions;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script.Config
+{
+    /// <summary>
+    /// A hosting configuration file based <see cref="FileConfigurationProvider"/>.
+    /// </summary>
+    public class FunctionsHostingConfigProvider : FileConfigurationProvider
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FunctionsHostingConfigProvider"/> class.
+        /// </summary>
+        /// <param name="source">The source settings.</param>
+        public FunctionsHostingConfigProvider(FunctionsHostingConfigSource source) : base(source) { }
+
+        /// <summary>
+        /// Loads the hosting config data from a stream.
+        /// </summary>
+        /// <param name="stream">The stream to read.</param>
+        public override void Load(Stream stream)
+        {
+            StreamReader reader = new StreamReader(stream);
+            string text = reader.ReadToEnd();
+            Data = Parse(text);
+        }
+
+        private static Dictionary<string, string> Parse(string settings)
+        {
+            // Expected settings: "ENABLE_FEATUREX=1,A=B,TimeOut=123"
```

Comment:
Maybe in the future we can have a structured config file, like `ini` or `json`.

In the meantime, I recommend prefixing all the keys loaded with a well-known configuration section, so that options configuration can rely on section and not specific provider. A prefix like:

``` CSharp
.ToDictionary(a => $"FunctionsPlatform:{a[0]}", a => a[1], StringComparer.OrdinalIgnoreCase);
```